includes/classes/InternalConnections/NetworkSiteConnection.php

```diff
@@ -832,12 +832,13 @@ public static function build_available_authorized_sites( $user_id = false, $cont
 		$authorized_sites = get_transient( $cache_key );
 
 		if ( $force || false === $authorized_sites ) {
-			$sites           = get_sites(
+			$authorized_sites = ! is_array( $authorized_sites ) ? array() : $authorized_sites;
```

Comment:
```suggestion
			$authorized_sites = array();
```

I think it should always start empty:
* `=== false` it's not an array
* `$force === true` then we're repopulating the authorized sites from 0. If the old value is retained then each site will appear in the array twice.

Reply:
Yeah, good point, I think this should always be a new empty array if we make it inside this conditional. I've made that change